src/modules/generators/open_simplex.rs

```diff
@@ -0,0 +1,632 @@
+// Copyright (c) 2017 The Noise-rs Developers.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT
+// or http://opensource.org/licenses/MIT>, at your option. All files in the
+// project carrying such notice may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Note that this is NOT Ken Perlin's simplex noise, as that is patent encumbered.
+//! Instead, these functions use the OpenSimplex algorithm, as detailed here:
+//! http://uniblock.tumblr.com/post/97868843242/noise
+
+use {PermutationTable, gradient, math};
+use math::{Point2, Point3, Point4};
+use modules::{NoiseModule, Seedable};
+use num_traits::Float;
+use std::ops::Add;
+
+const STRETCH_CONSTANT_2D: f64 = -0.211324865405187; //(1/sqrt(2+1)-1)/2;
+const SQUISH_CONSTANT_2D: f64 = 0.366025403784439; //(sqrt(2+1)-1)/2;
+const STRETCH_CONSTANT_3D: f64 = -1.0 / 6.0; //(1/Math.sqrt(3+1)-1)/3;
+const SQUISH_CONSTANT_3D: f64 = 1.0 / 3.0; //(Math.sqrt(3+1)-1)/3;
+const STRETCH_CONSTANT_4D: f64 = -0.138196601125011; //(Math.sqrt(4+1)-1)/4;
+const SQUISH_CONSTANT_4D: f64 = 0.309016994374947; //(Math.sqrt(4+1)-1)/4;
+
+const NORM_CONSTANT_2D: f32 = 1.0 / 14.0;
+const NORM_CONSTANT_3D: f32 = 1.0 / 14.0;
+const NORM_CONSTANT_4D: f32 = 1.0 / 6.8699090070956625;
+
+pub const DEFAULT_OPENSIMPLEX_SEED: usize = 0;
+
+/// Noise module that outputs 2/3/4-dimensional Open Simplex noise.
+#[derive(Clone, Copy, Debug)]
+pub struct OpenSimplex {
+    seed: usize,
```

Comment:
I was actually going to make another PR removing all public seeds, because it allows people to break invariants. Since you can update the seed without updating the corresponding permutations table, you can technically fail to seed something if you manually set the seed, then call set_seed with the same value (since it'll early exit)

This could be ignored, or it could be assumed that most of the objects will be immutable so this doesn't matter. I'll implement whichever is decided upon :)